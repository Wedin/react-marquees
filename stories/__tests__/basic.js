import React from "react";
import TestRenderer from "react-test-renderer";

import Marquee from "../../src/";

function createNodeMock(element) {
  if (element.type === "div") {
    return {
      parentNode: {
        getBoundingClientRect: () => ({
          bottom: 0,
          height: 200,
          left: 0,
          right: 0,
          top: 0,
          width: 200,
          x: 50,
          y: 25,
        }),
      },
    };
  }
  return null;
}

it("should render", () => {
  const options = { createNodeMock };

  const result = TestRenderer.create(
    <Marquee direction="left">
      <p>OK</p>
    </Marquee>,
    options
  ).toJSON();
  expect(result).toMatchSnapshot();
});
