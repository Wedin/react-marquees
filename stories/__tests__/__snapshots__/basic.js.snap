// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <div>
     
    <p>
      OK
    </p>
  </div>
</div>
`;
